DuckDuckGo/OnboardingContainerView.swift

```diff
@@ -0,0 +1,149 @@
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import SwiftUI
+
+#if APP_TRACKING_PROTECTION
+
+struct OnboardingButtonStyle: ButtonStyle {
+    func makeBody(configuration: Self.Configuration) -> some View {
+        configuration.label
+            .frame(maxWidth: .infinity, minHeight: Const.Size.buttonHeight, maxHeight: Const.Size.buttonHeight)
+            .background(configuration.isPressed ? Color.blue70 : Color(designSystemColor: .accent))
+            .cornerRadius(Const.Size.cornerRadius)
+    }
+}
+
+struct OnboardingContainerView: View {
+    
+    let viewModels: [OnboardingStepViewModel]
+    let enableAppTP: () -> Void
+    
+    @Binding var isLoading: Bool
+    
+    @State var currentModel: Int = 0
+    
+    func nextModel() {
+        withAnimation {
+            currentModel += 1
+        }
+    }
+    
+    func finishOnboarding() {
+        enableAppTP()
+    }
+    
+    private var isPad: Bool {
+        return UIDevice.current.userInterfaceIdiom == .pad
+    }
+    
+    var body: some View {
+        VStack {
+            let currentViewModel = viewModels[currentModel]
+            OnboardingStepView(viewModel: currentViewModel)
+                .padding(
+                    .horizontal,
+                    isPad ? Const.Size.horizontalStackPaddingPad : Const.Size.horizontalStackPadding
+                )
+            
+            Spacer()
+            
+            Button(action: {
+                if currentViewModel != viewModels.last {
+                    nextModel()
+                } else {
+                    finishOnboarding()
+                }
+            }, label: {
+                if isLoading {
+                    if #available(iOS 16, *) {
+                        SwiftUI.ProgressView()
+                            .tint(Color.buttonLabelColor)
+                    } else {
+                        SwiftUI.ProgressView()
+                            .progressViewStyle(CircularProgressViewStyle(tint: Color.buttonLabelColor))
+                    }
+                } else {
+                    Text(currentViewModel.primaryButtonTitle)
+                        .font(Font(uiFont: Const.Font.buttonFont))
+                        .foregroundColor(Color.buttonLabelColor)
+                }
+            })
+            .buttonStyle(OnboardingButtonStyle())
+            .padding(.bottom, Const.Size.buttonPadding)
+            .padding(
+                .horizontal,
+                isPad ? Const.Size.buttonHPaddingPad : Const.Size.buttonHPadding
+            )
+        }
+        .padding(.vertical)
+        .background(Color(designSystemColor: .surface))
+        .ignoresSafeArea()
+    }
+}
+
+private enum Const {
+    enum Font {
+        static let titleFont = UIFont.boldAppFont(ofSize: 28)
```

Comment:
As other comments, these should come from design system.  You can `import DesignResourcesKit` and see what's available in there.  If there's stuff in Figma that isn't in there, please push back on the designer.